research-software-directory/RSD-as-a-service
# feat: scrape organisation location from ror

# Adds ROR scraper for organisation's city and country

Fixes #806

Changes proposed in this pull request:

* adds a new scraper that collects country and city from ROR and stores them in the database
* the scraper takes organisations that do have a ROR id, but do not have a country OR a city and overwrites both information with data from ROR

How to test:

* optional: add `MAX_REQUESTS_ROR` to `.env`
* `docker compose down --volumes && docker compose build --parallel && docker compose up`
* login, create a new software and add a couple of organisations
* wait a few minutes for the scraper to run or run:
  ``` 
  docker compose exec scrapers /opt/java/openjdk/bin/java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.ror.MainRor
  ```
* verify that countries appear inside cards of the organisation overview

PR Checklist:
